### PR TITLE
fuzz, flamenco: collect rent for fee payer separate from other accounts

### PR DESCRIPTION
**Pull request by mjain-jump**
Also remove accounts with invalid fee payers (fee payer account was a builtin account which should not be able to pass sigverify)

### CODE REVIEW

File: src/flamenco/runtime/context/fd_exec_txn_ctx.c [hunk 1]
```diff
@@ -151,11 +152,14 @@ fd_txn_borrowed_account_modify_idx( fd_exec_txn_ctx_t *        ctx,
   }
 
   fd_borrowed_account_t * txn_account = &ctx->borrowed_accounts[idx];
+  if( !fd_txn_account_is_writable_idx( ctx, (int)idx ) ) {
```
Comment (ibhatt-jumptrading) [inline src/flamenco/runtime/context/fd_exec_txn_ctx.c:155]: UNLIKELY.

File: src/flamenco/runtime/fd_executor.c [hunk 2]
```diff
@@ -355,12 +355,13 @@ fd_executor_load_transaction_accounts( fd_exec_txn_ctx_t * txn_ctx ) {
...
 
-    /* Try to collect rent on all writable accounts. If rent is collected
+    /* Try to collect rent on all writable accounts, except the fee payer (who's rent was already
+       collected within fd_executor_validate_transaction_fee_payer()). If rent is collected
        successfully, update the starting lamports accordingly to avoid
        unbalanced lamports issues during instruction execution.
        TODO: the rent epoch check in the conditional should probably be moved
        to inside fd_runtime_collect_rent_from_account. */
-    if( fd_txn_account_is_writable_idx( txn_ctx, (int)i ) && acct->const_meta->info.rent_epoch<=epoch ) {
+    if( i>0 && fd_txn_account_is_writable_idx( txn_ctx, (int)i ) && acct->const_meta->info.rent_epoch<=epoch ) {
```
Comment (ibhatt-jumptrading) [inline src/flamenco/runtime/fd_executor.c:364]: why dont you just loop starting at i=1

File: src/flamenco/runtime/context/fd_exec_txn_ctx.c (hunk 1 above)
Comment (ibhatt-jumptrading) [reply src/flamenco/runtime/context/fd_exec_txn_ctx.c:155]: I don't love how we handle the borrowed account modify semantics... Probably should be addressed in a different PR

**Review by ibhatt-jumptrading — COMMENTED** [review]
annmoying change... sucks that it has to get done

File: src/flamenco/runtime/fd_executor.c (hunk 2 above)
Comment (mjain-jump) [reply src/flamenco/runtime/fd_executor.c:364]: initially bc of the account data size calculations, but upon second thought we can start at 1 cuz fee payer dlen should always be 0

**Review by ibhatt-jumptrading — APPROVED** [review]

**Pull request merged**